src/Sarif/Core/Run.cs

```diff
@@ -13,6 +14,109 @@ public partial class Run
         private static Invocation EmptyInvocation = new Invocation();
         private static LogicalLocation EmptyLogicalLocation = new LogicalLocation();
 
+        private IDictionary<FileLocation, int> _fileToIndexMap;
+
+        public int GetFileIndex(
+            FileLocation fileLocation, 
+            bool addToFilesTableIfNotPresent = true,
+            OptionallyEmittedData dataToInsert = OptionallyEmittedData.None,
+            System.Text.Encoding encoding = null)
+        {
+            if (fileLocation == null) throw new ArgumentNullException(nameof(fileLocation));
+
+            if (this.Files == null || this.Files.Count == 0)
+            {
+                return -1;
+            }
+
+            if (_fileToIndexMap == null)
+            {
+                InitializeFileToIndexMap();
+            }
+
+            // Strictly speaking, some elements that may contribute to a files table 
+            // key are case sensitive, e.g., everything but the schema and protocol of a
+            // web URI. We don't have a proper comparer implementation that can handle 
+            // all cases. For now, we cover the Windows happy path, which assumes that
+            // most URIs in log files are file paths (which are case-insensitive)
+            //
+            // Tracking item for an improved comparer:
+            // https://github.com/Microsoft/sarif-sdk/issues/973
+
+            // When we perform a files table look-up, only the uri and uriBaseId
+            // are relevant; these properties together comprise the unique identity
+            // of the file object. The file index, of course, does not relate to the
+            // file identity. We consciously exclude the properties as well.
```

Comment:
>. We consciously exclude the properties as well [](start = 28, length = 47)

"..., so we exclude it" ? Or "..., so we exclude that property?" Not sure what to make of the plural here.

Reply:
we exclude the properties _bag_.

---
In reply to: [244521805](https://github.com/Microsoft/sarif-sdk/pull/1191#discussion_r244521805) [](ancestors = 244521805)